compiler-core/src/error.rs

```diff
@@ -1842,13 +1842,18 @@ constructing a new record with its values."
                     location,
                     variables,
                     name,
-                    type_with_name_in_scope,
+                    situation
                 } => {
-                    let text = if *type_with_name_in_scope {
-                        wrap_format!("`{name}` is a type, it cannot be used as a value.")
-                    } else {
-                        wrap_format!("The name `{name}` is not in scope here.")
+                    let text = match situation {
+                        Some(UnknownValueConstructorErrorSituation::TypeWithNameIsInScope) => {
+                            wrap_format!("`{name}` is a type, it cannot be used as a value.")
+                        }
+                        Some(UnknownValueConstructorErrorSituation::NameStartsWithCapitalCase) => {
+                            wrap_format!("The name `{name}` is not in scope here.\n\nThis pattern is matching a constructor named `{name}`, but there is no constructor in scope with that name.")
+                        }
+                        _ =>  wrap_format!("The name `{name}` is not in scope here.")
```

Comment:
We don't use discard patterns ever as they disable exhaustiveness checking.
```suggestion
                        None =>  wrap_format!("The name `{name}` is not in scope here.")
```